sympy/solvers/inequalities.py

```diff
@@ -982,3 +986,489 @@ def reduce_inequalities(inequalities, symbols=[]):
+
+    Parameters
+    ==========
+
+    Under the maximization interpretation, m is the number of constraints and n
+    is the number of variables.
+
+    A : Matrix with shape (m, n)
+        Must contain only Rational and float coefficients.
+
+    B : Matrix with shape (m, 1)
+        Must contain only Rational and float coefficients.
+
+    C: Matrix with shape (1, n)
+        Must contain only Rational and float coefficients.
+
+    Returns
+    =======
+
+    Returns a triplet of solutions optimum, argmax, and argmax_dual:
+
+    optimum : float or Rational
+        maximum value of objective function under the constraints
+
+    argmax : list of floats and/or Rationals
+        x
+
+    argmax_dual : list of floats and/or Rationals
+        y
+
+    Examples
+    ========
+
+    Suppose we want to maximize
+        3x+y
+    subject to
+        x+y <= 2
+
+    >>> from sympy.matrices.dense import Matrix
+    >>> from sympy.solvers.inequalities import linprog_from_matricies
+    >>> A = Matrix([[1, 1]])
+    >>> B = Matrix([[2]])
+    >>> C = Matrix([[3, 1]])
+    >>> optimum, argmax, argmax_dual = linprog_from_matricies(A, B, C)
+    >>> optimum
+    6
+    >>> argmax
+    [2, 0]
+    >>> argmax_dual
+    [3]
+
+    Suppose we want to maximize
+        3x+y
+    subject to
+        x <= 2
+        y <= 4
+
+    >>> A = Matrix([[1,0], [0, 1]])
+    >>> B = Matrix([[2],[4]])
+    >>> C = Matrix([[3, 1]])
+    >>> optimum, argmax, argmax_dual = linprog_from_matricies(A, B, C)
+    >>> optimum
+    10
+    >>> argmax
+    [2, 4]
+    >>> argmax_dual
+    [3, 1]
+
+    See Also
+    ========
+
+    linprog_maximize
+    find_feasible
+    """
+    return _simplex(A, B, C)
+
+
+def _linear_programming_to_matrix(constraints, objective):
+    """
+    Converts a list of constraints and an objective function into the standard
+    form for linear programming. Each free variable is assigned a column
+    lexicographically.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.inequalities import _linear_programming_to_matrix
+    >>> from sympy.abc import x, y, z
+    >>> from sympy import Eq
+    >>> r1 = y+2*z <= 3
+    >>> r2 = -x-3*z <= -2
+    >>> r3 = 2*x+y+7*z <= 5
+    >>> A, B, C, variables, constraints = _linear_programming_to_matrix([r1, r2, r3], x + y + 5*z)
+    >>> A
+    Matrix([
+    [ 0, 1,  2],
+    [-1, 0, -3],
+    [ 2, 1,  7]])
+    >>> B
+    Matrix([
+    [ 3],
+    [-2],
+    [ 5]])
+    >>> C
+    Matrix([[1, 1, 5]])
+    >>> variables
+    [x, y, z]
+    >>> constraints
+    [y + 2*z <= 3, -x - 3*z <= -2, 2*x + y + 7*z <= 5]
+    >>> A, B, C, variables, constraints = _linear_programming_to_matrix([Eq(x, 3)], x*10)  # x = 3 become x >= 3 and x <= 3
+    >>> A
+    Matrix([
+    [ 1],
+    [-1]])
+    >>> B
+    Matrix([
+    [ 3],
+    [-3]])
+    >>> C
+    Matrix([[10]])
+    >>> variables
+    [x]
+    >>> constraints
+    [x <= 3, -x <= -3]
+    """
+    variables = objective.free_symbols
```

Comment:
I still think that using free symbols here is a mistake. The user should provide the symbols. Here is a concrete example of why:
```python
In [2]: f, g, h = symbols('f g h', cls=Function)

In [3]: objective = f(x) + 2*g(x) - h(x)

In [4]: objective
Out[4]: f(x) + 2⋅g(x) - h(x)

In [5]: objective.free_symbols
Out[5]: {x}
```
We might want to maximise over expressions like `f(x), g(x), h(x)` rather than plain symbols.